pycheribuild/config/compilation_targets.py

```diff
@@ -211,17 +216,37 @@ def riscv_arch_string(self):
     def riscv_abi(self):
         assert self.target.is_riscv(include_purecap=True)
         if self.target.is_cheri_purecap():
```

Comment:
```suggestion
        if self.is_baremetal():
            return self.riscv_softfloat_abi()  # Baremetal/FreeRTOS only supports softfloat
        if self.target.is_cheri_purecap():
```
I think you need something like this to make the arch string and the ABI match up.